airflow/sensors/external_task.py

```diff
@@ -40,6 +40,8 @@ class ExternalTaskSensorLink(BaseOperatorLink):
     name = 'External DAG'
 
     def get_link(self, operator, dttm):
+        ti = TaskInstance(task=operator, execution_date=dttm)
+        operator.render_template_fields({"ti": ti})
```

Comment:
Hmm, we should try to discourage creating ad-hoc TaskInstance objects since it is much too brittle now that it has a foreign key to DagRun. Not sure what’s a good alternative though. @ashb Maybe this is a good time to (finally) start working on a new interface for this?

Also instead of just `{"ti": ti}` we should likely use `ti.get_template_context()`.

Reply:
Ah OK. Took a look at the other operator links too and all but one creates an ad hoc TaskInstance, but honestly those probably could just use `XCom.get_one()`.

I'll update to use `ti.get_template_context()` in the meantime and test.